test/unit/host_test.rb

```diff
@@ -1854,6 +1854,11 @@ def to_managed!
     assert_equal [pclass.name], host.individual_puppetclasses.map(&:name)
   end
 
+  test "host created with config group updates the correct count" do
+    host = FactoryGirl.create(:host, :with_config_group)
+    assert_equal host.config_groups.last.hosts_count, 1
```

Comment:
expected, actual.

I'd suggest asserting the value of hostgroups_count to ensure it didn't increment.